doc/making-a-visualizer.md

```diff
@@ -20,187 +214,78 @@ the following data and methods:
-
-The other arguments in the export module are the visualizer itself
-(`Differences` in the example case, which is the name of the class we
-created), and a description.
-
-If you place the file containing your visualizer class definition with the
-export module in the folder name `Visualizers`, the engine will automatically
-package it up and include it in the list of available visualizers.
-
-There is no compiling needed. Simply place your file in the appropriate folder
-and run the app with `npm run dev`. The JavaScript will be compiled at
+   details, and you can just implement the `checkParameters()` method, which
+   only has to return a `ValidationStatus` that indicates whether the
+   parameter values are sensible.
+6. `show()`: Start or resume display of the visualization.
+7. `stop()`: Pause display of the visualization. Don't erase any visualization
+   produced so far or otherwise clean up the visualizer.
+8. `depart()`: Throw out the visualization, release its resources, remove its
+   injected DOM elements, and do any other required cleanup. After this call,
+   the visualizer must support `inhabit()` being called again, perhaps with a
+   different div, to re-initialize the visualization.
+
+### Where to put your visualizer
+
+When a visualizer is ready for Numberscope users, place the file containing
+its class definition and export module in the folder `src/visualizers`. When
+the Frontscope client runs, it'll find your visualizer and compile it at
 runtime.
 
-## Handling errors
+The visualizers in `src/visualizers` can be tested with the usual
+`npm run dev` call.
+
+As discussed [earlier](#test-your-visualizer-on-the-workbench), visualizers
+that aren't ready for Numberscope users should go in
+`src/visualizers-workbench`. You can load them for testing by calling
+`npm run dev:workbench`.
+
+### How to document your visualizer
+
+The p5 Template visualizer and the Differences visualizer both follow our
+documentation conventions.
+
+#### Name, thumbnail, and description
+
+In a documentation block just before the visualizer class definition, put a
+level-1 heading with the visualizer's name, followed by a thumbnail image and
+a description.
+
+To make the thumbnail, take a screenshot of your visualizer and put it in the
+`src/assets/img<VisualzierName>` directory. Set the `width` property in the
+`img` tag so that the image displays with a height of 320&nbsp;px or less.
+(You'll have to calculate the necessary width; setting the `height` property
+won't work.) Put the image tag inside a link to the image file.
+
+Below the thumbnail, describe what the visualizer does.
+
+### Parameters
+
+In a documentation block just before definition of `params`, put "Parameters"
+in a level-2 heading. Inside the definition of `params`, just before each the
+parameter's key, put a documentation block that displays like this:
+
+> **Parameter name:** Description of the parameter. _(Type and constraints.)_
+
```

Comment:
Remember to update this if we decide on different formatting.

Reply:
Are we planning to change the formatting in this PR? If not, what action should we take to remember?

Reply:
Hmm, did you see my comments/question about whether we needed italics for that bit of a parameter description? Parentheses might already be enough. I was just worried that this formatting was getting a bit too "stylified" for optimal ease of reading. In some cases, depending on the parameter constraints, it can get to be italics. Can't recall if I had any other formatting queries/thoughts in that comment. I just felt it was one of the unresolved design points that this PR has come to comprise and that we needed to reach consensus on the presentation here.

Reply:
> Hmm, did you see my comments/question about whether we needed italics for that bit of a parameter description?

Oh, no, I didn't see that.

> I was just worried that this formatting was getting a bit too "stylified" for optimal ease of reading.

I agree that the italics are a bit harder to read, but I like the way they clearly separate constraints. I was following the way JSDoc uses three distinct formatting modes for argument names, types, and descriptions.

Reply:
As mentioned it some other comment, Kate can eyeball the graphic design later. Resolving for now.